query-request.hh

```diff
@@ -44,6 +44,7 @@ using range = wrapping_range<T>;
 
 using ring_position = dht::ring_position;
 using clustering_range = nonwrapping_range<clustering_key_prefix>;
+clustering_range reverse(const clustering_range& range);
```

Comment:
I'm gonna paste a comment by Avi from my PR:
>  Please add a comment that explains that the reversed interval must be used with a comparator that has the reverse sense of the one that was used to interpret the input.

(https://github.com/scylladb/scylla/pull/9281/#discussion_r702404085)

For example: suppose we have a range `r`, point `p`, and comparator `cmp`.
To check if `p` is inside `r` we can use `r.contains(p, cmp)`.
Now it does not make sense to call `reverse(r).contains(p, cmp)`. The reversed range is supposed to be used with a reversed comparator: `reverse(r).contains(p, reverse(cmp))`, where `reverse(cmp)(a, b) = cmp(b, a)`.

Even more specific example, which is our intended use case, is where `cmp` is the schema comparator. The reversed ranges are supposed to be used with comparators created from the reversed schema. It would be good to mention this example in a comment here.